### ISSUE TITLE
How to add multiple principals or account ids to aws_ram_principal_association

### ISSUE THREAD

**Issue by medavamshi** [hashicorp/terraform-provider-aws#13365]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_XXXXX

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by medavamshi** [comment]
Error: No RAM Resource Share (arn:aws:ram:us-east-1:XXXXXXXXXXXX:resource-share/b84ab2b9-41ac-4c80-9ff2-10c275bfb8dd) invitation found
I get this error when i try to add multiple accounts one after the other as associations to transit gateway using RAM.  A second or third attempt works fine.  Is this due to dependency ?   what are the options to specify multiple accounts as principals at once?

**Comment by justinretzolk** [comment]
Hey @medavamshi 👋 Thank you for taking the time to file this issue. It looks like it may have been corrected with #17032; can you confirm whether you're still experiencing this issue after that PR was merged and released?
